datasets/str_sa.py

```diff
@@ -0,0 +1,80 @@
+from os import listdir
+from os.path import isfile
+
+from chipbase import Chip
+from chipbase import ChipBase
+
+
+class STR_SA(ChipBase):
+
+    def __init__(self, directory, dataset_name="STR_SA", *args, **kwargs):
+        super().__init__(dataset_name, args, kwargs)
+
+        # Get some chips
+        self.__load_chips(directory)
+
+    def __load_chips(self, directory):
+        self.chips = {}
+
+        for file in listdir(directory):
+            path = directory + '/' + file
+
+            # Only interested in certain files
+            is_valid = isfile(path)
+            is_png = path.endswith(".png")
+            is_mask = "mask" in path
+            if not is_valid or not is_png or is_mask:
+                continue
+
+            # Set all Chip variables
+            car_id = get_sa_car_id(path)
+            cam_id = get_sa_cam_id(path)
+            chip_id = str(car_id) + "_" + str(cam_id)
+            time = cam_id  # Cars always pass the first camera first
+            misc = None  # No misc data
+
+            # Make chip
+            chip = Chip(
+                chip_id,
+                car_id,
+                cam_id,
+                time,
+                path,
+                misc,
+            )
+
+            self.chips[chip_id] = chip
+
+    def get_all_chips_by_carid(self, carid):
+        return [chip for chip in self.chips.values() if chip.car_id == carid]
+
+    def get_all_chips_by_camid(self, camid):
+        return [chip for chip in self.chips.values() if chip.camid == camid]
+
+    def get_chip_image_path(self, chip_id):
+        chip = self.chips.get(chip_id, None)
+        if chip is not None:
+            return chip.filename
+        else:
+            return None
+
+
+def int_from_string(string, start_chars, int_len):
```

Comment:
Agreed, I can remove these if we make the base class `self.chips` a dictionary. It currently assumes a list.